src/cluster_resources.rs

```diff
@@ -30,7 +30,7 @@ use tracing::{debug, info};
 /// implementations and removes the orphaned resources. Therefore if a new implementation is added,
 /// it must be added to [`ClusterResources::delete_orphaned_resources`] as well.
 pub trait ClusterResource:
-    Clone + Debug + DeserializeOwned + Resource<DynamicType = ()> + Serialize
+    Clone + Debug + DeserializeOwned + Resource<DynamicType = ()> + GetApi + Serialize
```

Comment:
`ClusterResources` should only be used for namespaced resources because the orphanity check does not work for cluster-scoped resources which could be shared between product clusters. So `ClusterResource` can be restricted to namespaced resources:

```suggestion
    Clone + Debug + DeserializeOwned + Resource<DynamicType = ()> + GetApi<Namespace = str> + Serialize
```

This eliminates the need for `GetApi::namespace_from_str` within operator-rs.

Reply:
61ed83fee7c51949a170834886f3efe271638e19